action.yml

```diff
@@ -57,19 +58,78 @@ runs:
             brew install gmp zlib && brew link zlib
           fi
         shell: bash
-      - name: "Compile GAP and download packages"
+
+      - name: "Download CI scripts"
+        shell: bash
+        run:  git clone https://github.com/gap-system/pkg-ci-scripts.git
+
+      - name: "Clone GAP"
+        shell: bash
+        run: git clone --depth=2 -b ${{ inputs.GAPBRANCH }} https://github.com/gap-system/gap.git $HOME/gap
+
+      - name: "Build GAP and download packages"
+        shell: bash
+        env:
+          ABI: ${{ inputs.ABI }}
+        run: |
+          cd $HOME/gap
+          
+          # for HPC-GAP, add suitable flags
+          if [[ ${{ inputs.HPCGAP }} = yes ]]; then
+            GAP_CONFIGFLAGS="$GAP_CONFIGFLAGS --enable-hpcgap"
+          fi
+          
+          # build GAP in a subdirectory
+          ./autogen.sh
+          ./configure $GAP_CONFIGFLAGS
+          make -j4 V=1
+          
+          # download packages; instruct wget to retry several times if the
+          # connection is refused, to work around intermittent failures
+          make bootstrap-pkg-${{ inputs.GAP_BOOTSTRAP }} WGET="wget -N --no-check-certificate --tries=5 --waitretry=5 --retry-connrefused"
+          
+      - name: "Clone additional GAP packages"
+        shell: bash
+        run: |
+          # optionally: clone specific package versions, in case we need to test
+          # with development versions
+          cd $HOME/gap/pkg
+          for pkg in ${{ inputs.GAP_PKGS_TO_CLONE }}; do
+              # delete any existing variants of the package to avoid multiple versions
+              # from being compiled later on
+              rm -rf "$pkg"*
+              if [[ "$pkg" =~ ^http ]] ; then
+                  # looks like a full URL
+                  git clone "$pkg"
+              elif [[ "$pkg" =~ ^[^/]+/[^/]+$ ]] ; then
+                  # looks like ORG/REPO
+                  git clone https://github.com/"$pkg"
+              elif [[ "$pkg" =~ ^[^/]+$ ]] ; then
+                  # looks like just a REPO name
+                  git clone https://github.com/gap-packages/"$pkg"
+              else
+                  echo "Invalid package name or URL '$pkg' in GAP_PKGS_TO_CLONE"
+                  exit 1
+              fi
+          done
+          
+      - name: "Build additional GAP packages"
+        shell:  bash
+        env:
+          ABI: ${{ inputs.ABI }}
         run: |
+          # build some packages; default is to build 'io' and 'profiling', in order to
+          # generate coverage results.
```

Comment:
See also issue #26 (but of course: one step at a time!)

Reply:
I agree with #26!